mslib/plot.py

```diff
@@ -65,8 +69,10 @@ def load_from_xml_data(datasource):
         else:
             comments = ''
 
-        waypoints_list.append((lat, lon, flightlevel, location, comments))
-    return waypoints_list
+        waypoints_list1.append((lat, lon, flightlevel, location, comments))
+        waypoints_list2.append(ft.Waypoint(lat, lon, flightlevel, location, comments))
```

Comment:
As most of the mslib stuff uses the list of ft.Waypoint's, I would try to remove the need of the waypoints_list1.